atc/db/worker_task_cache.go

```diff
@@ -0,0 +1,111 @@
+package db
+
+import (
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/concourse/concourse/atc"
+)
+
+type WorkerTaskCache struct {
+	WorkerName string
+	TaskCache  UsedTaskCache
+}
+
+type UsedWorkerTaskCache struct {
+	ID         int
+	WorkerName string
+}
+
+func (wtc WorkerTaskCache) findOrCreate(
+	tx Tx,
+) (*UsedWorkerTaskCache, error) {
+	uwtc, found, err := wtc.find(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	if !found {
+		var id int
+		err = psql.Insert("worker_task_caches").
+			Columns(
+				"worker_name",
+				"task_cache_id",
+			).
+			Values(wtc.WorkerName, wtc.TaskCache.ID()).
+			Suffix(`
+					ON CONFLICT (worker_name, task_cache_id) DO UPDATE SET
+						task_cache_id = ?
+					RETURNING id
+				`, id).
```

Comment:
🤔This should probably be wtc.TaskCache.ID(), or you can also just use the EXCLUDED.task_cache_id instead of passing in an param so that it'll use the id value of the task_cache_id it conflicted with.

Reply:
Good catch thx! There was a block before this to look for task cache by sql and was returning `id`. Now it is using the find() method instead.